configure.ac

```diff
@@ -80,6 +84,40 @@ AC_SEARCH_LIBS([lrint], [m], [
 ])
 pkg_libs="$LIBS"
 
+AS_IF([test "x$enable_pthreads" = xyes && test "x$enable_w32threads" = xyes], [
+    AC_MSG_ERROR([both pthreads and w32threads were requested.])
+])
+
+AS_IF([(test "x$enable_pthreads" = xyes || test "x$enable_w32threads" = xyes) && test "x$ac_cv_header_stdatomic_h" != xyes], [
+    AC_MSG_ERROR([thread support was requested, but <stdatomic.h> was not found.])
+])
+
+threads_found=false
+
+AS_IF([test "x$enable_w32threads" != xno && test "x$ac_cv_header_stdatomic_h" = xyes], [
+    AS_IF([test "x$ac_cv_header_windows_h" = xyes], [
+        AC_DEFINE(CONFIG_W32THREAD, 1, [Have Win32 threads])
+        threads_found=true
+    ], [test "x$enable_w32threads" = xyes], [
+        AC_MSG_ERROR([win32 threading support was requested, but was not found.])
+    ])
+])
+
+AS_IF([test "x$threads_found" != xtrue && test "x$enable_pthreads" != xno && test "x$ac_cv_header_stdatomic_h" = xyes], [
+    AX_PTHREAD([
+        pkg_libs="$pkg_libs $PTHREAD_LIBS"
+        CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
+        LIBS="$LIBS $PTHREAD_LIBS"
+        CC="$PTHREAD_CC"
+        AC_DEFINE(CONFIG_PTHREAD, 1, [Have POSIX threads])
+        threads_found=true
+    ], [
+        AS_IF([test "x$enable_pthreads" = xyes], [
+            AC_MSG_ERROR([pthread support was requested, but was not found.])
+        ])
+    ])
+])
+
```

Comment:
nit: atm AX_PTHREAD produces some output indicating whether pthread was found, but if Win32 is used or there’s no `stdatomic.h` no indicators wrt threading are printed at all. Might be nice to add `AC_MSG_CHECKING` to the Win32 tests or alternatively print the overall result at the end:
```autoconf
thread_api="none" # instead of threads_found
# ....
AC_MSG_NOTICE([Chosen threading model: $thread_api])
```

Reply:
Sure, done.